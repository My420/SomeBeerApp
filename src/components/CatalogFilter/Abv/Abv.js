import React from 'react';
import styles from './Abv.module.scss';
import {
  ABV_LESS_PROP,
  ABV_MORE_PROP,
  ABV_BOTTOM_VALUE,
  ABV_TOP_VALUE
} from '../../../utils/constants';
import validateNumberValue from '../../../utils/validateNumberValue';

class Abv extends React.PureComponent {
  onABVChange = evt => {
    const { onPropertyChange } = this.props;
    const value = validateNumberValue(
      evt.target.value,
      ABV_BOTTOM_VALUE,
      ABV_TOP_VALUE
    );
    const direction = evt.target.name;
    if (direction === 'abvLess') {
      onPropertyChange(value, ABV_LESS_PROP);
    } else if (direction === 'abvMore') {
      onPropertyChange(value, ABV_MORE_PROP);
    }
  };

  render() {
    // eslint-disable-next-line no-console
    console.log('render ========== AbvInput');

    const { lessValue, moreValue } = this.props;
    return (
      <fieldset className={styles.abvField}>
        <legend className={styles.abvlegend}>Alcohol By Volume</legend>
        <label className={styles.abvLabelMore} htmlFor="abvMore">
          {'More'}
          <input
            id="abvMore"
            name="abvMore"
            className={styles.abvInputMore}
            type="number"
            value={moreValue}
            onChange={this.onABVChange}
          />
        </label>
        <label className={styles.abvLabelLess} htmlFor="abvLess">
          {'Less'}
          <input
            id="abvLess"
            name="abvLess"
            className={styles.abvInputLess}
            type="number"
            value={lessValue}
            onChange={this.onABVChange}
          />
        </label>
      </fieldset>
    );
  }
}

export default Abv;
