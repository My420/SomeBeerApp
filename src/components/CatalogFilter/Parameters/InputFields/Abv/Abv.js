import React from 'react';
import PropTypes from 'prop-types';
import styles from './Abv.module.scss';
import {
  ABV_LESS_PROP,
  ABV_MORE_PROP,
  ABV_BOTTOM_VALUE,
  ABV_TOP_VALUE
} from '../../../../../utils/constants';
import validateNumberValue from '../../../../../utils/validateNumberValue';

const Abv = ({ lessValue, moreValue, onPropertyChange }) => {
  const onABVChange = evt => {
    const value = validateNumberValue(
      evt.target.value,
      ABV_BOTTOM_VALUE,
      ABV_TOP_VALUE
    );
    const direction = evt.target.name;
    if (direction === 'abvLess') {
      onPropertyChange(value, ABV_LESS_PROP);
    } else if (direction === 'abvMore') {
      onPropertyChange(value, ABV_MORE_PROP);
    }
  };

  return (
    <fieldset className={styles.abvField}>
      <legend className={styles.abvLegend}>Alcohol By Volume</legend>
      <div className={styles.abvInner}>
        <label className={styles.abvLabelMore} htmlFor="abvMore">
          <span className={styles.textMore}>more than</span>
          <input
            id="abvMore"
            name="abvMore"
            className={styles.abvInputMore}
            type="number"
            value={moreValue}
            onChange={onABVChange}
          />
        </label>
        <label className={styles.abvLabelLess} htmlFor="abvLess">
          <span className={styles.textLess}>less than</span>
          <input
            id="abvLess"
            name="abvLess"
            className={styles.abvInputLess}
            type="number"
            value={lessValue}
            onChange={onABVChange}
          />
        </label>
      </div>
    </fieldset>
  );
};

Abv.propTypes = {
  lessValue: PropTypes.string.isRequired,
  moreValue: PropTypes.string.isRequired,
  onPropertyChange: PropTypes.func.isRequired
};

export default Abv;
