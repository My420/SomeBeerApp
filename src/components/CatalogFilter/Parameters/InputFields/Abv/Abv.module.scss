@import '../../../../../sass/variables.scss';

.abvField {
  padding-bottom: 10px;
  border: none;
  @include border-bottom;
}

.abvInputMore,
.abvInputLess {
  width: 40px;
  height: 25px;

  @include border;
  @include input;
}

.textMore,
.textLess {
  margin-right: 7px;
}

.abvInner {
  display: flex;
  justify-content: space-around;
}

.abvLegend {
  display: inline-block;
  text-align: center;
  font-weight: 700;
}

@media (max-width: 400px) {
  .abvField {
    padding-bottom: 5px;
  }

  .abvInputMore,
  .abvInputLess {
    width: 30px;
    height: 20px;
    font-size: 14px;
  }
}
