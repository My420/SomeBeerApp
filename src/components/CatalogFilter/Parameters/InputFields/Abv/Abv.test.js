import React from 'react';
import { shallow } from 'enzyme';
import Abv from './Abv';

describe('test <Abv /> component', () => {
  const mockFunc = jest.fn();
  const wrapper = shallow(
    <Abv lessValue="20" moreValue="30" onPropertyChange={mockFunc} />
  );

  beforeEach(() => {
    mockFunc.mockReset();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render  inputs with attr name='abvLess' with correct value", () => {
    expect(wrapper.find("input[name='abvLess']").prop('value')).toBe('20');
  });

  test("should render  inputs with attr name='abvMore' with correct value", () => {
    expect(wrapper.find("input[name='abvMore']").prop('value')).toBe('30');
  });

  test("should call onPropertyChange prop when inputs with attr name='abvLess' change", () => {
    wrapper
      .find("input[name='abvLess']")
      .simulate('change', { target: { value: '40', name: 'abvLess' } });
    expect(mockFunc.mock.calls.length).toBe(1);
    expect(mockFunc.mock.calls[0][0]).toBe('40');
  });

  test("should call onPropertyChange prop when inputs with attr name='abvMore' change", () => {
    wrapper
      .find("input[name='abvMore']")
      .simulate('change', { target: { value: '10', name: 'abvMore' } });
    expect(mockFunc.mock.calls.length).toBe(1);
    expect(mockFunc.mock.calls[0][0]).toBe('10');
  });
});
