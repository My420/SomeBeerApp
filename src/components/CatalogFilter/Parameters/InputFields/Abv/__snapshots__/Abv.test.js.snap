// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <Abv /> component should match snapshot 1`] = `
<fieldset
  className="abvField"
>
  <legend
    className="abvLegend"
  >
    Alcohol By Volume
  </legend>
  <div
    className="abvInner"
  >
    <label
      className="abvLabelMore"
      htmlFor="abvMore"
    >
      <span
        className="textMore"
      >
        more than
      </span>
      <input
        className="abvInputMore"
        id="abvMore"
        name="abvMore"
        onChange={[Function]}
        type="number"
        value="30"
      />
    </label>
    <label
      className="abvLabelLess"
      htmlFor="abvLess"
    >
      <span
        className="textLess"
      >
        less than
      </span>
      <input
        className="abvInputLess"
        id="abvLess"
        name="abvLess"
        onChange={[Function]}
        type="number"
        value="20"
      />
    </label>
  </div>
</fieldset>
`;
